Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java

```diff
@@ -89,38 +86,55 @@ public boolean isExpired(Date invocationDate) {
      * @return true if the timestamp is before (now-timeToLive), false otherwise
      */
     public boolean verifyCreated(int timeToLive, int futureTimeToLive, Date invocationDate) {
-        if (invocationDate == null) {
-            invocationDate = new Date();
-        }
+        Date date = invocationDate == null ? new Date() : invocationDate;
 
-        long invocationTime = invocationDate.getTime();
+        long invocationTime = date.getTime();
         if (futureTimeToLive > 0) {
-            invocationDate.setTime(invocationTime + futureTimeToLive * 1000);
+            date.setTime(invocationTime + futureTimeToLive * 1000);
```

Comment:
I don't follow the logic here.  Wouldn't the setTime on line 93 always be overridden by setTime on line 103?